spell/psm-checklists.md

```diff
@@ -0,0 +1,113 @@
+# PSM Checklists
+
+## PSM Migration Checklist
+
+- [ ] IF contracts are being on-boarded in the current spell, execute the full [LitePSM On-boarding Checklist](#litepsm-on-boarding-checklist)
+- [ ] IF a PSM is deprecated in the migration, execute the full [PSM Off-boarding Checklist](#psm-off-boarding-checklist)
+- [ ] Amount of `gem` being migrated matches the Exec Sheet
+- [ ] Collateral amount that is fetched from the source PSM matches the collateral is added to the destination PSM
+- [ ] No bad debt is left behind
+- Source PSM:
+    - [ ] ilk matches source PSM
+    - [ ] IF present, `tin` matches the value in the Exec Sheet
+    - [ ] IF present, `tout` matches the value in the Exec Sheet
+    - [ ] `AutoLine` is updated according to the Exec Sheet
+- Destination PSM:
+    - [ ] ilk matches destination PSM
+    - [ ] IF present, `tin` matches the value in the Exec Sheet
+    - [ ] IF present, `tout` matches the value in the Exec Sheet
+    - [ ] `AutoLine` is updated according to the Exec Sheet
+    - [ ] `buf` matches the value in the Exec Sheet
+    - [ ] `fill` is called to ensure liquidity is immediately available after spell execution
+
+## LitePSM On-boarding Checklist
+
+- Deployed Contracts
+    - `DssLitePsm`
+        - [ ] Uses the agreed best Solidity compiler version at the time
+        - [ ] Optimizer is **enabled**
+        - [ ] Contract has been reviewed by at least a couple of DAO dev teams
+        - [ ] Contract has been audited by an external party
+            - [ ] Audit reports are public
+            - [ ] All issues found were fixed or acknowledged
+            - [ ] No code changes are done after final Audit report
```

Comment:
How do we verify that?

Reply:
There are different ways to carry out this task.

One of them would be to get the code at the latest commit in the audit report and manually diff-check them against Etherscan. However, this can be very tedious.

Here's how I would do it:

1. Clone the repo in my machine
2. 2. Get the latest commit hash in the relevant audit reports:
  ![image](https://github.com/user-attachments/assets/3a4c2345-51ad-4ff8-8a22-e24d5ae2d244)
3. Check out commit above:
    ```
    git checkout 2f11f4bc47d96f4ebf025a1e4a249987150f9baa
    ```
4. FOR EACH relevant contract, run `forge verify-bytecode`
    I.e.:
    ```bash
    forge verify-bytecode --rpc-url=$ETH_RPC_URL --etherscan-api-key=$ETHERSCAN_API_KEY \
      0xf6e72Db5454dd049d0788e411b06CfAF16853042 DssLitePsm \
      --constructor-args 0x4c4954452d50534d2d555344432d410000000000000000000000000000000000000000000000000000000000a0b86991c6218b36c1d19d4a2e9eb0ce3606eb480000000000000000000000009759a6ac90977b93b58547b4a71c78317f391a2800000000000000000000000037305b1cd40574e4c5ce33f8e8306be057fd7341
    ```
  > [!TIP]
  > You can get the encoded constructor args from Etherscan UI.
  
  > [!WARNING]
  > Make sure you have the latest version from Foundry, because there was a [bug](https://github.com/foundry-rs/foundry/issues/8255) affecting this command until very recently.

Reply:
I meant to suggest additional specific instructions into the checklist itself
```suggestion
            - [ ] No code changes are done after final Audit report (based on the commit hash found in the report + code diff check or bytecode verification tool such as `forge verify-bytecode`)
```

Reply:
Makes sense. Adding this.